executor/aggregate.go

```diff
@@ -45,39 +105,490 @@ type HashAggExec struct {
 	GroupByItems  []expression.Expression
 	groupKey      []byte
 	groupVals     [][]byte
+
+	// After we support parallel execution for aggregation functions with distinct,
+	// we can remove this attribute.
+	doesUnparallelExec bool
+
+	finishCh         chan struct{}
```

Comment:
There are so many channels here and in `HashAggPartialWorker`. Can you draw a graph to show how do them work?

Reply:
Added, PTAL
@shenli